lib/buffer.js

```diff
@@ -304,6 +306,20 @@ function slowToString(encoding, start, end) {
   if (end > this.length) end = this.length;
   if (end <= start) return '';
 
+  switch (encoding) {
+    case 'ucs2':
+    case 'ucs-2':
+    case 'utf16le':
+    case 'utf-16le':
+      if (((end - start) / 2) > kStringMaxLength)
+        throw new RangeError('invalid String length');
```

Comment:
I'm not sure how I feel about this.  I'm inclined to say that it should try to create the string and throw an exception when that fails.  @trevnorris WDYT?

Aside: you could write this as `if ((end - start) > (kStringMaxLength >>> 1))`, V8's optimizing compiler can probably constant-fold that.